--- erts/etc/common/Makefile.in.orig	2008-03-27 13:11:52.000000000 +0300
+++ erts/etc/common/Makefile.in	2008-03-27 13:13:23.000000000 +0300
@@ -298,7 +298,7 @@
 	$(PURIFY) $(LD) $(LDFLAGS) $(ENTRY_LDFLAGS) -o $@ $(OBJDIR)/inet_gethost.o $(ENTRY_OBJ) $(LIBS) 
 
 $(BINDIR)/run_erl: $(OBJDIR)/run_erl.o
-	$(LD) $(LDFLAGS) -o $@ $(OBJDIR)/run_erl.o $(LIBS)
+	$(LD) $(LDFLAGS) -o $@ $(OBJDIR)/run_erl.o $(LIBS) -lutil
 
 $(OBJDIR)/run_erl.o: ../unix/run_erl.c
 	$(CC) $(CFLAGS) -o $@ -c ../unix/run_erl.c
