diff -up otp_src_R12B-3/lib/ssl/c_src/Makefile.in.sslrpath otp_src_R12B-3/lib/ssl/c_src/Makefile.in
--- otp_src_R12B-3/lib/ssl/c_src/Makefile.in.sslrpath	2008-09-05 22:01:58.000000000 +0200
+++ otp_src_R12B-3/lib/ssl/c_src/Makefile.in	2008-09-05 22:02:34.000000000 +0200
@@ -102,7 +102,7 @@ else
 SSL_MAKEFILE =
 endif
 
-CC_R_FLAG=@CFLAG_RUNTIME_LIBRARY_PATH@
+CC_R_FLAG=
 ifeq ($(findstring @,$(CC_R_FLAG)),@)
 # Old erts configure used which hasn't replaced @CFLAG_RUNTIME_LIBRARY_PATH@;
 # we try our best here instead...
