--- otp_src_R11B-2/erts/emulator/hipe/hipe_x86_signal.c.glibc25	2006-12-30 23:57:04.000000000 +0100
+++ otp_src_R11B-2/erts/emulator/hipe/hipe_x86_signal.c	2006-12-30 23:57:36.000000000 +0100
@@ -34,7 +34,7 @@
 #endif
 #include "hipe_signal.h"
 
-#if __GLIBC__ == 2 && (__GLIBC_MINOR__ == 3 || __GLIBC_MINOR__ == 4)
+#if __GLIBC__ == 2 && (__GLIBC_MINOR__ == 3 || __GLIBC_MINOR__ == 4 || __GLIBC_MINOR__ == 5)
 /* See comment below for glibc 2.2. */
 #ifndef __USE_GNU
 #define __USE_GNU		/* to un-hide RTLD_NEXT */
