diff -up otp_src_R12B-0/lib/orber/c_src/setsockaddr_in.c.null otp_src_R12B-0/lib/orber/c_src/setsockaddr_in.c
--- otp_src_R12B-0/lib/orber/c_src/setsockaddr_in.c.null	2008-02-23 13:46:52.000000000 +0100
+++ otp_src_R12B-0/lib/orber/c_src/setsockaddr_in.c	2008-02-23 13:47:15.000000000 +0100
@@ -17,6 +17,7 @@
  *     $Id: otp-null.patch,v 1.1 2008/02/23 13:41:16 gemi Exp $
  *
  */
+#include <stdlib.h>
 #include "inet.h"
 
 #ifdef HAVE_CONFIG_H
