diff -up otp_src_R13B02/erts/etc/unix/Install.src.install otp_src_R13B02/erts/etc/unix/Install.src
--- otp_src_R13B02/erts/etc/unix/Install.src.install	2009-09-18 16:07:12.000000000 +0200
+++ otp_src_R13B02/erts/etc/unix/Install.src	2009-10-22 15:49:01.000000000 +0200
@@ -111,7 +111,7 @@ if [ -h epmd ]; then
   /bin/rm -f epmd
 fi
 
-ln -s $TARGET_ERL_ROOT/erts-%I_VSN%/bin/epmd epmd
+ln -s ../erts-%I_VSN%/bin/epmd epmd
 
 cp -p $ERL_ROOT/erts-%I_VSN%/bin/run_erl .
 cp -p $ERL_ROOT/erts-%I_VSN%/bin/to_erl .
@@ -160,16 +160,4 @@ if [ "X$TARGET" != "Xsunos5" -a -d $ERL_
 	(ranlib $library) > /dev/null 2>&1
     done
 fi
-
-
-#
-# Fixing the man pages
-#
-
-if [ -d $ERL_ROOT/man ]
-then
-    cd $ERL_ROOT
-    ./misc/format_man_pages $ERL_ROOT
-fi
-
-
+exit 0
