--- otp_src_R10B-8/lib/crypto/c_src/Makefile.in.rpath	2005-10-29 22:50:59.000000000 +0200
+++ otp_src_R10B-8/lib/crypto/c_src/Makefile.in	2005-10-29 22:51:23.000000000 +0200
@@ -79,7 +79,7 @@
 ifeq ($(HOST_OS),)
 HOST_OS := $(shell $(ERL_TOP)/erts/autoconf/config.guess)
 endif
-LD_R_FLAG=@DED_LD_FLAG_RUNTIME_LIBRARY_PATH@
+LD_R_FLAG=
 ifeq ($(strip $(LD_R_FLAG)),)
 LD_R_OPT =
 else
--- otp_src_R10B-8/lib/crypto/priv/Makefile.rpath	2005-10-29 22:52:01.000000000 +0200
+++ otp_src_R10B-8/lib/crypto/priv/Makefile	2005-10-29 22:52:32.000000000 +0200
@@ -60,7 +60,7 @@
 # ----------------------------------------------------
 
 $(SO_DRIVER): $(OBJS)
-	$(SO_LD) $(SO_LDFLAGS) -L$(SO_SSL_LIBDIR) -Wl,-R$(SO_SSL_LIBDIR) \
+	$(SO_LD) $(SO_LDFLAGS) -L$(SO_SSL_LIBDIR) \
 	-o $@ $^ -lcrypto
 
 $(DLL_DRIVER): $(OBJS)
--- otp_src_R10B-8/lib/ssl/c_src/Makefile.in.rpath	2005-10-29 22:53:15.000000000 +0200
+++ otp_src_R10B-8/lib/ssl/c_src/Makefile.in	2005-10-29 22:53:50.000000000 +0200
@@ -104,9 +104,9 @@
 CC_R_FLAG =
 else
 ifeq ($(findstring osf,$(TARGET)),osf)		# osf1: -Wl,-rpath,
-CC_R_FLAG = -Wl,-rpath,
+CC_R_FLAG =
 else						# Default: -Wl,-R
-CC_R_FLAG = -Wl,-R
+CC_R_FLAG =
 endif
 endif
 
